front-end/src/components/UserForm/SignIn.tsx

```diff
@@ -7,26 +7,53 @@ interface UserForm {
 interface State {
     isSubmitButtonEnable: boolean,
     username: string,
+    showBtnInPasswordInput: boolean,
+    isShowingPassword: boolean,
 }
 
 class SignIn extends React.Component<UserForm, State>{
 
     state = {
         isSubmitButtonEnable: false,
-        username: ""
+        username: "",
+        showBtnInPasswordInput: false,
+        isShowingPassword: false
     }
 
-    handleUserNameChange = (e: { target: { value: string; }; }) => {
+    handleUserNameChange = (e: { target: { value: string; }; }): void => {
```

Comment:
and this should be a functional setState to make sure we have the latest update in chain

Reply:
You use a functional setSate If the next state depends on the current state. I think we do not have  the need to use a functional setSate here. Please Take a look at: https://reactjs.org/docs/react-component.html#setstate, and let me know what you think.